simulator/modules/branch/branch.h

```diff
@@ -42,9 +41,19 @@ class Branch : public Log
     public:
         explicit Branch( bool log);
         void clock( Cycle cycle);
-        auto get_mispredict_rate() const { return ( num_branches != 0) 
-                                                  ? num_mispredictions / num_branches * 100
-                                                  : 0; }
+        auto get_mispredictions_num() const { return num_mispredictions; }
+
+        bool is_misprediction( const Instr& instr, const BPInterface& bp_data) const
+        {
+            bool is_misprediction = false;
+
+            if ( bp_data.is_hit)
+                is_misprediction |= ( bp_data.is_taken != instr.is_taken());
+
+            is_misprediction |= bp_data.is_taken && ( bp_data.target != instr.get_new_PC());
```

Comment:
Same here — we may just return here.
One of the guidelines of clean code is to avoid assignments :-)